.pre-commit-config.yaml

```diff
@@ -29,4 +29,4 @@ repos:
           - mmh3
           - semver
           - panzi-json-logic
-        exclude: proto|tests
+        exclude: providers/openfeature-provider-flagd|tests
```

Comment:
I'm sorry, but I'm not really familiar with the tooling, so I will most likely ask some questions out of curiosity. Why do we want to exclude the provider from mypy? especially as we add it to the type checking in the build.yml

Reply:
because I'm moving away from a central `mypy` approach to a per package. the other ones will follow after this PR 🙂

Reply:
Understood, but I would suggest splitting this into a separate and dedicated PR - Depending on the discussions, it can easily be that one or the other topic in this pr might lead to some controversy, and hence, one or the other improvement will be blocked by something unrelated. Furthermore adds this complexity to a topic, and for others it might be hard to understand what is now related to actually schema migration topic, and what are build improvements.